source/framefmt.c

```diff
@@ -235,16 +252,23 @@ int aws_cryptosdk_serialize_frame(
     size_t plaintext_size,
     struct aws_byte_buf *ciphertext_buf,
     const struct aws_cryptosdk_alg_properties *alg_props) {
+    AWS_PRECONDITION(aws_cryptosdk_alg_properties_is_valid(alg_props));
     struct aws_cryptosdk_framestate state;
 
+    // The plaintext_size should be bound to prevent arithmetic
+    // overflows due to addition
+    if (plaintext_size > MAX_PLAINTEXT_SIZE) {
```

Comment:
The `plaintext_size` should be bounded because otherwise there could be an overflow in lines 170, 194, 206.

Reply:
Agreed, but see my comments about frame size limits.